src/components/Channels/Channels.js

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { fetchChannels } from "../../utils/fetcher";
+import Messages from "../Messages/Messages";
+
+class Channels extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      channels: [],
+      currentChannel: "General",
+      error: null
+    };
+  }
+
+  componentDidMount() {
+    let data = fetchChannels();
+    this.setState({ channels: data, currentChannel: "General" });
+  }
+
+  handleClick = e => {
+    e.preventDefault();
+    this.setState({ currentChannel: e });
```

Comment:
This won't do what you think it will. The parameter `e` is an Event, you're passing in an Event object for a string.